test/standalone/Bucket/snow_bucket_tests.jl

```diff
@@ -54,6 +54,107 @@ for FT in (Float32, Float64)
+
+        T_atmos = (t) -> 280.0
+        u_atmos = (t) -> 10.0
+        q_atmos = (t) -> 0.03
+        h_atmos = FT(3)
+        P_atmos = (t) -> 101325 # Pa
+        bucket_atmos = PrescribedAtmosphere(
+            TimeVaryingInput(liquid_precip),
+            TimeVaryingInput(snow_precip),
+            TimeVaryingInput(T_atmos),
+            TimeVaryingInput(u_atmos),
+            TimeVaryingInput(q_atmos),
+            TimeVaryingInput(P_atmos),
+            ref_time,
+            h_atmos,
+        )
+
+        τc = FT(10.0)
+        surface_space = d.space.surface
+        albedo =
+            BulkAlbedoFunction{FT}(α_snow, α_bareground_func, surface_space)
+        bucket_parameters = BucketModelParameters(
+            κ_soil,
+            ρc_soil,
+            albedo,
+            σS_c,
+            W_f,
+            z_0m,
+            z_0b,
+            τc,
+            earth_param_set,
+        )
+        model = BucketModel(
+            parameters = bucket_parameters,
+            domain = d,
+            atmosphere = bucket_atmos,
+            radiation = bucket_rad,
+        )
+
+        # run for some time
+        Y, p, coords = initialize(model)
+        Y.bucket.T .= init_temp.(coords.subsurface.z, FT(274.0))
+        Y.bucket.W .= -0.1 # small negative moisture
+        Y.bucket.Ws .= 0.0 # no runoff
+        Y.bucket.σS .= -0.1 # small negative snow
+        t0 = FT(0.0)
+        dY = similar(Y)
+
+        exp_tendency! = make_exp_tendency(model)
+        set_initial_cache! = make_set_initial_cache(model)
+        set_initial_cache!(p, Y, t0)
+        # Test that the albedo is the bareground albedo (negative snow treated as zero snow)
+        @test p.bucket.α_sfc ==
+              FT.(
+            α_bareground_func.(
+                ClimaCore.Fields.coordinate_field(surface_space)
+            )
+        )
+        # Test that evaporation is zero
+        @test all(parent(p.bucket.turbulent_fluxes.vapor_flux) .== FT(0.0))
+        # Test that q_sfc is computed as if there was zero snow
+        @test p.bucket.q_sfc ==
+              ClimaLand.Bucket.saturation_specific_humidity.(
+            p.bucket.T_sfc,
+            FT(0.0),
+            p.bucket.ρ_sfc,
+            earth_param_set.thermo_params,
+        )
+        # Test that beta factor is zero despite negative water content
+        @test all(
+            parent(
+                ClimaLand.Bucket.beta_factor.(
+                    Y.bucket.W,
+                    Y.bucket.σS,
+                    model.parameters.W_f,
+                    model.parameters.σS_c,
+                ),
+            ) .== FT(0.0),
+        )
+        exp_tendency!(dY, Y, p, t0)
+        # Test that all water tendencies are zero (no precip, no snowmelt, no vapor loss)
+        @test dY.bucket.W == ClimaCore.Fields.zeros(surface_space)
+        @test dY.bucket.Ws == ClimaCore.Fields.zeros(surface_space)
+        @test dY.bucket.σS == ClimaCore.Fields.zeros(surface_space)
+    end
+
```

Comment:
may it be worth adding a test for the `partition_surface_fluxes` as well, just to ensure all fluxes are the expected signs?

Reply:
added some in!